src/jenkins_job_insight/main.py

```
@@ -2383,17 +2474,108 @@ async def api_dashboard() -> list[dict]:
+
+@app.post("/api/validate-token")
+async def validate_token(
+    body: ValidateTokenRequest,
+    settings: Settings = Depends(get_settings),
+) -> dict:
+    """Validate a GitHub or Jira token by making a lightweight API call.
+
+    GitHub: GET /user (returns authenticated user info)
+    Jira: GET /rest/api/2/myself (returns authenticated user info)
+    """
+    if body.token_type == "github":
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to: "token_type"

(S105)